packer/scripts/Artifacts/MsSql/Structure/Ods/2050-CreateTableValueFunctions.sql

```diff
@@ -0,0 +1,574 @@
+    INNER JOIN reporting.GetFullTimeEquivalency(@searchDate) med ON
+        ssa.StudentUSI = med.StudentUSI
+        AND sch_Reporting.LocalEducationAgencyId = med.LocalEducationAgencyId
+    INNER JOIN edfi.Descriptor d_gradelevel ON
+        ssa.EntryGradeLevelDescriptorId = d_gradelevel.DescriptorId
+    WHERE
+        med.total_fte >= CASE WHEN d_gradelevel.CodeValue IN ('hp', 'pk') THEN 0 else 51 END
+        AND (ssa.ExitWithdrawDate IS NULL OR ssa.ExitWithdrawDate >= @searchDate)
+        AND ssa.EntryDate <= @searchDate
+    GROUP BY
+        ssa.StudentUSI,
+        med.LocalEducationAgencyId,
+        total_fte;
+
+    RETURN;
+END;
+GO
+
+CREATE OR ALTER FUNCTION reporting.GetMinimalSchoolReportingId
+(
+    @searchDate DATE
+)
+RETURNS @ResultTable TABLE
+(
+    LocalEducationAgencyId INT,
+    StudentUSI INT,
+    Total_FTE INT,
+    ReportingSchoolId INT
+    INDEX idx_minSchoolId CLUSTERED (LocalEducationAgencyId, StudentUSI)
+)
+AS
+BEGIN
+    INSERT INTO @ResultTable (StudentUSI, LocalEducationAgencyId, Total_FTE, ReportingSchoolId)
+    SELECT
+        ssa.StudentUSI,
+        fte.LocalEducationAgencyId,
+        Total_FTE,
+        ReportingSchoolId = MIN(ssae.ReportingSchoolId)
+    FROM edfi.StudentSchoolAssociation ssa
+    INNER JOIN sk.StudentSchoolAssociationExtension ssae ON
+        ssa.StudentUSI = ssae.StudentUSI
+        AND ssa.SchoolId = ssae.SchoolId
+        AND ssa.EntryDate = ssae.EntryDate
+    INNER JOIN edfi.School sch_Reporting ON
+        ssae.ReportingSchoolId = sch_Reporting.SchoolId
+    INNER JOIN reporting.GetMaxFullTimeEquivalency(@searchDate) fte ON
+        ssa.StudentUSI = fte.StudentUSI
+        AND sch_Reporting.LocalEducationAgencyId = fte.LocalEducationAgencyId
+        AND ssa.FullTimeEquivalency = fte.Max_FTE
+    WHERE
+        (ssa.ExitWithdrawDate IS NULL or ssa.ExitWithdrawDate >= @searchDate)
+        AND ssa.EntryDate <= @searchDate
+    GROUP BY ssa.StudentUSI, fte.LocalEducationAgencyId, total_fte;
+
+    RETURN;
+END;
+GO
+
+CREATE OR ALTER FUNCTION reporting.GetHomeLanguageCode
+(
+)
+RETURNS @ResultTable TABLE
+(
+    StudentUSI INT,
+    EducationOrganizationId INT,
+    CodeValue NVARCHAR(50),
+    Description NVARCHAR(1024),
+    INDEX idx_homelanguage CLUSTERED (StudentUSI, EducationOrganizationId)
+)
+AS
+BEGIN
+    INSERT INTO @ResultTable (StudentUSI, EducationOrganizationId, CodeValue, Description)
+    SELECT
+        seod.StudentUSI,
+        seod.EducationOrganizationId,
+        d.CodeValue,
+        d.Description
+    FROM edfi.StudentEducationOrganizationAssociation seod WITH (NOLOCK)
+    INNER JOIN edfi.StudentEducationOrganizationAssociationLanguage seodl WITH (NOLOCK) ON
+        seod.StudentUSI = seodl.StudentUSI
+        AND seod.EducationOrganizationId = seodl.EducationOrganizationId
+    INNER JOIN edfi.StudentEducationOrganizationAssociationLanguageUse seodlu WITH (NOLOCK) ON --I think most records are missing this, it isn't required?
```

Comment:
is this comment still relevant? 
```suggestion
    INNER JOIN edfi.StudentEducationOrganizationAssociationLanguageUse seodlu WITH (NOLOCK) ON
```